src/Css.elm

```diff
@@ -13525,3 +13537,1002 @@ pointerEvents :
     -> Style
 pointerEvents (Value val) =
     AppendProperty ("pointer-events:" ++ val)
+
+
+
+--- Scroll ---
+
+
+{-| Sets `smooth` value for usage with [`scrollBehavior`](#scrollBehavior).
+
+    scrollBehavior smooth
+
+-}
+smooth : Value { provides | smooth : Supported }
+smooth =
+    Value "smooth"
+
+
+{-| Sets [`scroll-behavior`](https://css-tricks.com/almanac/properties/s/scroll-behavior/)
+
+    scrollBehavior auto
+
+    scrollBehavior smooth
+
+-}
+scrollBehavior :
+    Value
+        { auto : Supported
+        , smooth : Supported
+        , inherit : Supported
+        , initial : Supported
+        , unset : Supported
+        }
+    -> Style
+scrollBehavior (Value val) =
+    AppendProperty ("scroll-behavior:" ++ val)
+
+
+{-| Sets [`scroll-margin`](https://css-tricks.com/almanac/properties/s/scroll-margin/) property.
+
+    scrollMargin (em 4)
+
+    scrollMargin2 (em 4) (px 2)
+
+    scrollMargin3 (em 4) (px 2) (pct 5)
+
+    scrollMargin4 (em 4) (px 2) (pct 5) (px 3)
+
+-}
```

Comment:
I saw an elm-css user the other day getting frustrated because the docs for margin were:

```elm
margin (px 10)
margin2 (px 10) (px 10)
margin3 (px 10) (px 10) (px 10)
margin4 (px 10) (px 10) (px 10) (px 10)
```

These have kind of the same problem—they show some allowable types but are not really useful as a reference to remember how the behavior works. I think we could do a little better here. What about documenting what each of the positions do? Does it go clockwise from the top like elsewhere in scrollMargin4? top/bottom and left/right in scrollMargin2? What's the actual behavior in the 3-value  case?

Reply:
I've updated the documentation. I hope it clarifies a lot. I can add the same documentation to `margin` and `padding` if you want?